record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
Remind me why this is necessary? Do you expect to be able to recover if you hit ErrUnexpectedEOF? Doesn't that mean that there isn't any next chunk?